runtime-modules/project-token/src/types.rs

```diff
@@ -0,0 +1,337 @@
+use codec::{Decode, Encode};
+use frame_support::{
+    dispatch::{fmt::Debug, DispatchResult},
+    ensure,
+};
+use sp_arithmetic::traits::{Saturating, Zero};
+use sp_runtime::traits::{Convert, Hash};
+use sp_std::collections::btree_map::{BTreeMap, Iter};
+use sp_std::iter::Sum;
+
+/// Info for the account
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub struct AccountData<Balance> {
+    /// Non-reserved part of the balance. There may still be restrictions
+    /// on this, but it is the total pool what may in principle be
+    /// transferred, reserved and used for tipping.
+    pub(crate) free_balance: Balance,
+
+    /// This balance is a 'reserve' balance that other subsystems use
+    /// in order to set aside tokens that are still 'owned' by the
+    /// account holder, but which are not usable in any case.
+    pub(crate) reserved_balance: Balance,
+}
+
+/// Info for the token
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, Debug)]
+pub struct TokenData<Balance, Hash, BlockNumber> {
+    /// Current token issuance
+    pub(crate) supply: Balance,
+
+    /// Initial issuance state
+    pub(crate) issuance_state: OfferingState,
+
+    /// Transfer policy
+    pub(crate) transfer_policy: TransferPolicy<Hash>,
+
+    /// Symbol used to identify token
+    pub(crate) symbol: Hash,
+
+    /// Patronage Information
+    pub(crate) patronage_info: PatronageData<Balance, BlockNumber>,
+}
+
+/// Patronage information, patronage configuration = set of values for its fields
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Default, Debug)]
+pub struct PatronageData<Balance, BlockNumber> {
+    /// Patronage rate
+    pub(crate) rate: Balance,
```

Comment:
This was due to the latest change:
##### Before
`patronage_amount = blocks * rate`
##### Now
`patronage_amount = blocks * rate * supply`

Reply:
So according to the comment it looks to me that you need to:
- Introduce module type/constant like `PatronageRateNumberOfBlocks` and set it to `5256000` (blocks / year assuming 6s target block time)
- `Permill` then should be sufficient for `rate`
- Calculate the unclaimed amount with: `(current_block - last_tally_update_block) / PatronageRateNumberOfBlocks * rate * supply + tally` (just make sure it's never rounded up to avoid exceeding the actual rate)